common/src/main/java/org/jboss/hal/testsuite/testlistener/ServerStateListenerOperations.java

```diff
@@ -0,0 +1,124 @@
+ * Created by pjelinek on Apr 19, 2016
+ */
+public class ServerStateListenerOperations {
+
+    public Map<String, Map<String, String>> getServerStateMap() throws IOException {
+        return tryWithClient(client -> {
+            ModelReader reader = new ModelReader(client);
+            return reader.getHostStream().collect(toMap(identity(), reader.getServerOfHostStatesFunction()));
+        });
+    }
+
+    public boolean isReloadRequired() throws IOException {
+        return tryWithClient(client -> new Administration(client).isReloadRequired());
+    }
+
+    public boolean isRestartRequired() throws IOException {
+        return tryWithClient(client -> new Administration(client).isRestartRequired());
+    }
+
+    private <T> T tryWithClient(Function<OnlineManagementClient, T> modelReadFunction) throws IOException {
+        try (OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient()) {
+            return modelReadFunction.apply(client);
+        }
+    }
+
+    private class ModelReader {
+
+        private ModelReader(OnlineManagementClient client) {
+            this.ops = new Operations(client);
+        }
+
+        private Operations ops;
+
+        /**
+         * {@link Function} instance to return map of server -> status pairs for given host
+         */
+        private Function<String, Map<String, String>> getServerOfHostStatesFunction() {
+            return host -> {
+                return getServerStream(host).collect(toMap(identity(), server -> {
+                    return readServerConfigStatus(host, server);
+                }));
+            };
+        }
+
+        private Stream<String> getHostStream() {
+            ModelNodeResult hostsResult = null;
+            try {
+                hostsResult = ops.readChildrenNames(Address.root(), HOST);
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to read hosts.", e);
+            }
+            hostsResult.assertDefinedValue();
+            return hostsResult.stringListValue().stream();
+        }
+
+        private Stream<String> getServerStream(String host) {
+            ModelNodeResult serversResult;
+            try {
+                serversResult = ops.readChildrenNames(Address.host(host), SERVER);
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to read servers of host.", e);
+            }
+            serversResult.assertDefinedValue();
+            return serversResult.stringListValue().stream();
+        }
+
+        private String readServerConfigStatus(String host, String server) {
+            ModelNodeResult serverConfigStatusResult;
+            try {
+                serverConfigStatusResult = ops.readAttribute(Address.host(host).and(SERVER_CONFIG, server), STATUS);
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to read server config status.", e);
```

Comment:
Yes it's done this way to avoid the necessity of polluting lambda inside `Function` instance with try catch block which would be necessary otherwise since `apply` method of `Function` don't allow to rethrow any checked exception.